cura/Settings/IntentManager.py

```diff
@@ -0,0 +1,141 @@
+            #       even though there should always be defaults. The problem then is what to do with the quality_types.
+            #       Currently _also_ inconsistent with 'currentAvailableIntentCategoreis', which _does_ return default.
+        quality_groups = application.getQualityManager().getQualityGroups(global_stack)
+        available_quality_types = {quality_group.quality_type for quality_group in quality_groups.values() if quality_group.node_for_global is not None}
+
+        final_intent_ids = set() #type: Set[str]
+        current_definition_id = global_stack.definition.getMetaDataEntry("id")
+        for extruder_stack in ExtruderManager.getInstance().getUsedExtruderStacks():
+            nozzle_name = extruder_stack.variant.getMetaDataEntry("name")
+            material_id = extruder_stack.material.getMetaDataEntry("base_file")
+            final_intent_ids |= {metadata["id"] for metadata in self.intentMetadatas(current_definition_id, nozzle_name, material_id) if metadata["quality_type"] in available_quality_types}
+
+        result = set() #type: Set[Tuple[str, str]]
+        for intent_id in final_intent_ids:
+            intent_metadata = application.getContainerRegistry().findContainersMetadata(id = intent_id)[0]
+            result.add((intent_metadata["intent_category"], intent_metadata["quality_type"]))
+        return list(result)
+
+    ##  List of intent categories available in either of the extruders.
+    #
+    #   This is purposefully inconsistent with the way that the quality types
+    #   are listed. The quality types will show all quality types available in
+    #   the printer using any configuration. This will only list the intent
+    #   categories that are available using the current configuration (but the
+    #   union over the extruders).
+    #   \return List of all categories in the current configurations of all
+    #   extruders.
+    def currentAvailableIntentCategories(self) -> List[str]:
+        global_stack = CuraApplication.getInstance().getGlobalContainerStack()
+        if global_stack is None:
+            return ["default"]
+        current_definition_id = global_stack.definition.getMetaDataEntry("id")
+        final_intent_categories = set() #type: Set[str]
+        for extruder_stack in ExtruderManager.getInstance().getUsedExtruderStacks():
+            nozzle_name = extruder_stack.variant.getMetaDataEntry("name")
+            material_id = extruder_stack.material.getMetaDataEntry("base_file")
+            final_intent_categories.update(self.intentCategories(current_definition_id, nozzle_name, material_id))
+        return list(final_intent_categories)
+
+    ##  The intent that gets selected by default when no intent is available for
+    #   the configuration, an extruder can't match the intent that the user
+    #   selects, or just when creating a new printer.
+    def defaultIntent(self) -> InstanceContainer:
```

Comment:
It's worded as a property but it's not. I think it's better to either change it to a property or word it as a function (getDefaultIntent)